src/types.ts

```diff
@@ -41,3 +41,13 @@ export interface RunnerResult {
     availableActions: string[]
   }
 }
+
+export type ParamsResult = {
+  templates: string
+  generator: string
+  action: string
+  subaction?: string
+  actionfolder?: string
+  name?: string
+  dry?: boolean
+} & object
```

Comment:
If there's a better way to do this, I'm open to suggestions. Still relatively new to TS.